src/utils/inject-process.js

```diff
@@ -1,16 +1,57 @@
 // @ts-nocheck
 // eslint-disable-next-line unicorn/prefer-module
 const _process = require('process/browser')
+const performance = globalThis.performance || {}
+const performanceNow =
+  performance.now ||
+  function () {
+    return Date.now()
+  }
 
+// generate timestamp or delta
+// see http://nodejs.org/api/process.html#process_process_hrtime
+function hrtime(previousTimestamp) {
+  const clocktime = performanceNow.call(performance) * 1e-3
+  let seconds = Math.floor(clocktime)
+  let nanoseconds = Math.floor((clocktime % 1) * 1e9)
+  if (previousTimestamp) {
+    seconds = seconds - previousTimestamp[0]
+    nanoseconds = nanoseconds - previousTimestamp[1]
+    if (nanoseconds < 0) {
+      seconds--
+      nanoseconds += 1e9
+    }
+  }
+  return [seconds, nanoseconds]
+}
 const p = {
   ..._process,
   exit: (code = 0) => {
     if (code === 0) {
-      globalThis.PW_TEST.end(false)
+      if (process.env.PW_TEST.mode === 'worker') {
+        postMessage({
+          pwRunEnded: true,
+          pwRunFailed: false,
+        })
+      } else {
+        globalThis.PW_TEST.end(false)
+      }
     } else {
-      globalThis.PW_TEST.end(true)
+      if (process.env.PW_TEST.mode === 'worker') {
+        postMessage({
+          pwRunEnded: true,
+          pwRunFailed: true,
+        })
+      } else {
+        globalThis.PW_TEST.end(true)
+      }
     }
   },
+  stdout: {
+    // eslint-disable-next-line no-console
+    write: console.log,
```

Comment:
Unrelated to changes here, but I would really like to be able to write output without added `\n` line delimiters. I was thinking maybe `stdout.write` could communicate with the host through postMessage or some other mechanism to do so. Alternatively logged messages could be decorated in some way on this side and than unwrapped on the other.

Alternatively compiled runner could be a module (as suggested earlier) with exported async iterable that takes care of running tests emitting messages that will be written into stdout. That would fit perfectly my test runner, but may not work for ones that utilize process.stdout

Reply:
can you make this a new issue please ?